packages/block-editor/src/components/block-rename/modal.js

```diff
@@ -23,14 +24,18 @@ export default function BlockRenameModal( {
 	originalBlockName,
 	onClose,
 	onSave,
+	hasOverridesWarning,
```

Comment:
I don't think the generic `block-editor` package shouldn't contain code that reasons about pattern overrides, which is a WordPress feature.

It's tricky because we're leveraging block naming, which is a generic block editor feature. 

I wondered if we could use an editor setting to disable the regular block naming UI and replace it with a custom one (possibly implemented in the `editor` package) when a user is editing a pattern.

Reply:
The more I think about it, the more I think this might be the simplest option 😅. The block-editor package also contains code for the block binding feature anyway. I think we can add a comment above it but maybe we don't need to extract it out, at least for now. WDYT?

Reply:
I haven't thought of any alternative, either.

> The block-editor package also contains code for the block binding feature anyway.

I guess block bindings could also be considered a general block editor feature, though an implementer would have to bring part of their own implementation (all the dynamic rewriting of blocks on the frontend). 🤔 

My view is that Custom Fields, Pattern Overrides are more the WordPress specific things built on top of it.

> I think we can add a comment above it but maybe we don't need to extract it out, at least for now. WDYT?

Making an issue for it, adding it to the main Tracking/Epic/Iteration for pattern overrides would be good as well as the comment. I think a comment is too easily forgotten. 😄

Reply:
Good point. Made an issue here: https://github.com/WordPress/gutenberg/issues/60619.